controllers/accesslogpolicy_controller.go

```diff
@@ -0,0 +1,206 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	anv1alpha1 "github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1"
+	"github.com/aws/aws-application-networking-k8s/pkg/aws"
+	"github.com/aws/aws-application-networking-k8s/pkg/config"
+	"github.com/aws/aws-application-networking-k8s/pkg/deploy"
+	"github.com/aws/aws-application-networking-k8s/pkg/k8s"
+	lattice_runtime "github.com/aws/aws-application-networking-k8s/pkg/runtime"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils/gwlog"
+)
+
+const (
+	accessLogPolicyFinalizer = "accesslogpolicy.k8s.aws/resources"
+)
+
+type accessLogPolicyReconciler struct {
+	log              gwlog.Logger
+	client           client.Client
+	scheme           *runtime.Scheme
+	finalizerManager k8s.FinalizerManager
+	eventRecorder    record.EventRecorder
+	stackDeployer    deploy.StackDeployer
+	cloud            aws.Cloud
+	stackMarshaller  deploy.StackMarshaller
+}
+
+func RegisterAccessLogPolicyController(
+	log gwlog.Logger,
+	cloud aws.Cloud,
+	finalizerManager k8s.FinalizerManager,
+	mgr ctrl.Manager,
+) error {
+	mgrClient := mgr.GetClient()
+	scheme := mgr.GetScheme()
+	evtRec := mgr.GetEventRecorderFor("accesslogpolicy")
+
+	stackDeployer := deploy.NewServiceNetworkStackDeployer(log, cloud, mgrClient)
+	stackMarshaller := deploy.NewDefaultStackMarshaller()
+
+	r := &accessLogPolicyReconciler{
+		log:              log,
+		client:           mgrClient,
+		scheme:           scheme,
+		finalizerManager: finalizerManager,
+		eventRecorder:    evtRec,
+		stackDeployer:    stackDeployer,
+		cloud:            cloud,
+		stackMarshaller:  stackMarshaller,
+	}
+
+	builder := ctrl.NewControllerManagedBy(mgr).
+		For(&anv1alpha1.AccessLogPolicy{})
+
+	return builder.Complete(r)
+}
+
+func (r *accessLogPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.log.Infow("reconcile", "name", req.Name)
+	recErr := r.reconcile(ctx, req)
+	res, retryErr := lattice_runtime.HandleReconcileError(recErr)
+	if res.RequeueAfter != 0 {
+		r.log.Infow("requeue request", "name", req.Name, "requeueAfter", res.RequeueAfter)
+	} else if res.Requeue {
+		r.log.Infow("requeue request", "name", req.Name)
+	} else if retryErr == nil {
+		r.log.Infow("reconciled", "name", req.Name)
+	}
+	return res, retryErr
+}
+
+func (r *accessLogPolicyReconciler) reconcile(ctx context.Context, req ctrl.Request) error {
+	alp := &anv1alpha1.AccessLogPolicy{}
+	if err := r.client.Get(ctx, req.NamespacedName, alp); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	if !alp.DeletionTimestamp.IsZero() {
+		return r.reconcileDelete(ctx, alp)
+	} else {
+		return r.reconcileUpsert(ctx, alp)
+	}
+}
+
+func (r *accessLogPolicyReconciler) reconcileDelete(ctx context.Context, alp *anv1alpha1.AccessLogPolicy) error {
```

Comment:
qq here: when you implement actual Access Log Subscription deletion in this method, how would you get the identifier to delete the access log?

Reply:
When we receive a delete, we can see the resource being deleted, which will allow us to see the target reference of Gateway or Route resource.

We can list the customer's VPC Lattice Service Networks or Services and find the one whose name matches the Kubernetes resource being created/deleted.

Once we have the id of the VPC Lattice resource, we can use it to call ListAccessLogSubscriptions, and delete the one whose destinationArn matches the deleted AccessLogPolicy destinationArn.

Reply:
Thanks